src/org/sosy_lab/java_smt/test/IntegerFormulaManagerTest.java

```diff
@@ -0,0 +1,33 @@
+package org.sosy_lab.java_smt.test;
+import java.math.BigDecimal;
+import org.junit.Test;
+import org.sosy_lab.java_smt.api.SolverException;
+import org.sosy_lab.java_smt.api.IntegerFormulaManager;
+import org.sosy_lab.java_smt.api.NumeralFormula.IntegerFormula;
+import org.sosy_lab.java_smt.api.BooleanFormulaManager;
+import org.sosy_lab.java_smt.api.BooleanFormula;
+
+public class IntegerFormulaManagerTest extends SolverBasedTest0 {
+
+  @Test
+  public void testBigDecimalInIntegerFormula() throws SolverException, InterruptedException {
+    // Test that BigDecimal values are handled correctly by all solvers
+    IntegerFormula f = imgr.makeNumber(BigDecimal.valueOf(3.5));
+    IntegerFormula three = imgr.makeNumber(3);
+    BooleanFormula equals = imgr.equal(f, three);
+    assertThatFormula(equals).isSatisfiable();
+  }
+  
+  @Test
+  public void testEuclideanDivisionForNegativeValues() throws SolverException, InterruptedException {
+    requireIntegers();
+    
+    // Test euclidean division for negative values with fractional parts
+    // For -3.5, we expect -4 (not -3 that simple truncation would give)
+    IntegerFormula f = imgr.makeNumber(BigDecimal.valueOf(-3.5));
+    IntegerFormula minusFour = imgr.makeNumber(-4);
+    
+    // This should be a tautology if euclidean division is working correctly
+    assertThatFormula(imgr.equal(f, minusFour)).isTautological();
+  }
+}
```

Comment:
The tests are missing several special cases:
- zero
- large numbers, greater than MAX_INT or MAX_LONG, or their negative pendant
- small numbers, smaller than MIN_FLOAT or MIN_DOUBLE

Additionally, one could write the test much more compact:
```
private void checkEqualInt(BigDecimal bd, BigInteger bi) {
  assertThatFormula(imgr.equal(mgr.makeNumber(bd), mgr.makeNumber(bi))).isTautological();
}

@Test
public void testIntegers() {
  checkEqualInt(BigDecimal.valueOf(0), BigInteger.valueOf(0));
  checkEqualInt(BigDecimal.valueOf(4), BigInteger.valueOf(4));
  checkEqualInt(BigDecimal.valueOf(4.5), BigInteger.valueOf(4));
  checkEqualInt(BigDecimal.valueOf(-4), BigInteger.valueOf(-4));
  checkEqualInt(BigDecimal.valueOf(-3.5), BigInteger.valueOf(-4));
  ...
}
```

Reply:
The test was not even running for all solvers, but only thee default SMTInterpol. Fixed in 531c545b7e24a1695352a1e56077e4453ba405fc .